### PR TITLE
feat(react): add locale hooks

### PR DESCRIPTION
**Pull request by andreribeirosilva**
## Description

This PR adds the following react hooks:

## useLocale

- Params
    - enableAutoFetch (optional)
- Return
    - data
        - subfolder (with stripped slash)
        - countryCultureCode
        - countryCode
        - sourceCountryCode
        - currency
        - country
            - code
            - structures
            - structure
            - currencies
            - cultures
            - … (country entity)

## useCountries

- Params
    - enableAutoFetch (optional)
- Return
    - isLoading
    - error
    - isFetched
    - data
        - countries
        - countriesById (object of countries indexed by countryId)
    - actions
        - fetch

## useCountryStates

- Params
    - countryCode
    - enableAutoFetch (optional)
- Return
    - isLoading
    - error
    - isFetched
    - data
        - countryStates (Array)
            - id
            - name
            - code
            - countryId
    - actions
        - fetch

## useCountryCities

- Params
    - countryCode
    - stateId
    - enableAutoFetch (optional)
- Return
    - osLoading
    - error
    - data
        - countryCities (Array)
            - id
            - name
            - countryId
            - stateId
    - actions
        - fetch

## useCountryAddressSchemas

- Params
    - enableAutoFetch (optional)
    - countryCode
- Return
    - isLoading
    - error
    - data
        - countriesAddressSchemas (all available schemas)
        - countryAddressSchemas
    - actions
        - fetch(countryCode)

### Dependencies

None.

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: packages/client/src/locale/types/country.ts [hunk 1]
```diff
@@ -16,6 +15,8 @@ export interface Country {
   cultures: string[];
   defaultSubfolder: string;
   defaultCulture: string;
+  states?: number[];
```
Comment (boliveira) [inline packages/client/src/locale/types/country.ts:18]: The `Country` type is defined for the client `getCountries` which does not have the `states` and `cities` in its response. Those properties are added only after normalization from the `fetchCountryStates` and `fetchCountryCities` actions, so it makes sense to have those properties defined in a type exclusive to the redux package. You can add those properties to the `CountriesEntity` which is located in `packages/redux/src/entities/types/countries.types.ts`.

File: packages/react/src/locale/hooks/useCountryAddressSchemas.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  StoreState,
+} from '@farfetch/blackout-redux';
+import { useAction } from '../../helpers';
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import type { UseCountryStatesOptions } from '../types/useCountryStates.types';
+
+export function useCountryAddressSchemas(
+  countryCode: string,
+  options: UseCountryStatesOptions = {},
```
Comment (boliveira) [inline packages/react/src/locale/hooks/useCountryAddressSchemas.ts:17]: What do you think of defining an alias for the `UseCountryStatesOptions` type like this:

```
export type UseCountryAddressSchemasOptions = UseCountryStatesOptions;
```

In that way, I believe the user would not be confused by the naming, as he might think it was a mistake.

Comment (andreribeirosilva) [reply packages/react/src/locale/hooks/useCountryAddressSchemas.ts:17]: Actually,  the type is created and I just put in place a wrong import. Nice catch. thanks 😉

File: packages/redux/src/locale/__tests__/selectors.test.ts [hunk 3]
```diff
@@ -192,6 +208,14 @@ describe('locale redux selectors', () => {
     });
   });
 
+  describe('getCity()', () => {
+    it('should get all the cities to a specific stateId', () => {
```
Comment (boliveira) [inline packages/redux/src/locale/__tests__/selectors.test.ts:212]: Is this test description correct?

File: packages/redux/src/locale/__tests__/selectors.test.ts [hunk 4]
```diff
@@ -200,6 +224,14 @@ describe('locale redux selectors', () => {
     });
   });
 
+  describe('getState()', () => {
+    it('should get all the cities to a specific stateId', () => {
```
Comment (boliveira) [inline packages/redux/src/locale/__tests__/selectors.test.ts:228]: Is this test description correct?

**Review by boliveira — APPROVED** [review]

**Pull request merged**